pkg/steps/cluster_claim.go

```diff
@@ -186,24 +200,23 @@ func mutate(secret *corev1.Secret, name, namespace string) (*corev1.Secret, erro
 	}, nil
 }
 
-func releaseCluster(ctx context.Context, hiveClient ctrlruntimeclient.Client, clusterClaim *hivev1.ClusterClaim, testName string) error {
-	logrus.Infof("Releasing cluster claims for test %s", testName)
+func (s *clusterClaimStep) releaseCluster(ctx context.Context, clusterClaim *hivev1.ClusterClaim) error {
+	logrus.Infof("Releasing cluster claims for test %s", s.Name())
 	logrus.WithField("clusterClaim.Namespace", clusterClaim.Namespace).WithField("clusterClaim.Name", clusterClaim.Name).Debug("Deleting cluster claim.")
-	if err := hiveClient.Delete(ctx, clusterClaim); err != nil {
+	if err := s.hiveClient.Delete(ctx, clusterClaim); err != nil {
```

Comment:
Likely worth it to retry a couple times on this?

Reply:
Add retry.
Why it could fail in the first place? Do we do the retry on other object deleting?